tests/zfs-tests/tests/functional/rsend/rsend.kshlib

```diff
@@ -551,3 +551,120 @@ function test_fs_setup
 	fi
 	log_must $ZFS create -o compress=lz4 $sendpool/stream
 }
+
+#
+# Check to see if the specified features are set in a send stream.
+# The values for these features are found in include/sys/zfs_ioctl.h
+#
+# $1 The stream file
+# $2-$n The flags expected in the stream
+#
+function stream_has_features
+{
+	typeset file=$1
+	shift
+
+	[[ -f $file ]] || log_fail "Couldn't find file: $file"
+	typeset flags=$($CAT $file | $ZSTREAMDUMP | $AWK '/features =/ {print $3}')
+	typeset -A feature
+	feature[dedup]="1"
+	feature[dedupprops]="2"
+	feature[sa_spill]="4"
+	feature[embed_data]="10000"
+	feature[lz4]="20000"
+	feature[mooch_byteswap]="40000"
+	feature[large_blocks]="80000"
+	feature[resuming]="100000"
+	feature[redacted]="200000"
+	feature[compressed]="400000"
+
+	typeset flag known derived=0
+	for flag in "$@"; do
+		known=${feature[$flag]}
+		[[ -z $known ]] && log_fail "Unknown feature: $flag"
+
+		derived=$($ECHO "$flags & ${feature[$flag]} = X" | mdb | $SED 's/ //g')
```

Comment:
I need a replacement for mdb here, I tried using ksh directly but it doesn't provide the correct output